src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -0,0 +1,245 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Invocation-side implementation of gRPC Asyncio Python."""
+import asyncio
+import enum
+from typing import Callable, Dict, Optional, ClassVar
+
+import grpc
+from grpc import _common
+from grpc._cython import cygrpc
+
+DeserializingFunction = Callable[[bytes], str]
+
+
+class AioRpcError(grpc.RpcError):
+    """An RpcError to be used by the asynchronous API."""
+
+    _code: grpc.StatusCode
+    _details: Optional[str]
+    _initial_metadata: Optional[Dict]
+    _trailing_metadata: Optional[Dict]
+
+    def __init__(self,
+                 code: grpc.StatusCode,
+                 details: Optional[str] = None,
+                 initial_metadata: Optional[Dict] = None,
+                 trailing_metadata: Optional[Dict] = None):
+        super().__init__(self)
+        self._code = code
+        self._details = details
+        self._initial_metadata = initial_metadata
+        self._trailing_metadata = trailing_metadata
+
+    def code(self) -> grpc.StatusCode:
+        """
+        Returns:
+          The `grpc.StatusCode` status code.
+        """
+        return self._code
+
+    def details(self) -> Optional[str]:
+        """
+        Returns:
+          The description of the error.
+        """
+        return self._details
+
+    def initial_metadata(self) -> Optional[Dict]:
+        """
+        Returns:
+          The inital metadata received.
+        """
+        return self._initial_metadata
+
+    def trailing_metadata(self) -> Optional[Dict]:
+        """
+        Returns:
+          The trailing metadata received.
+        """
+        return self._trailing_metadata
+
+
+@enum.unique
+class _RpcState(enum.Enum):
+    """Identifies the state of the RPC."""
+    ONGOING = 1
+    CANCELLED = 2
+    FINISHED = 3
+    ABORT = 4
+
+
+class Call:
+    """ Object for managing RPC calls,
+    returned when an instance of `UnaryUnaryMultiCallable` object is called.
+    """
+
+    _cancellation_details: ClassVar[str] = 'Locally cancelled by application!'
+
+    _state: _RpcState
+    _exception: Optional[Exception]
+    _response: Optional[bytes]
+    _code: grpc.StatusCode
+    _details: Optional[str]
+    _initial_metadata: Optional[Dict]
+    _trailing_metadata: Optional[Dict]
+    _call: asyncio.Task
+    _call_cancel_status: cygrpc.AioCancelStatus
+    _response_deserializer: DeserializingFunction
+
+    def __init__(self, call: asyncio.Task,
+                 response_deserializer: DeserializingFunction,
+                 call_cancel_status: cygrpc.AioCancelStatus) -> None:
```

Comment:
nit: Can you add a docstring to this function? The initialization arguments have specific expectations, and I hope we write it down (when memory still fresh).